### PR TITLE
US04 and US17 changes

### PR DESCRIPTION
**Pull request by rsukitha**
US04 and US17 changes checked in.

### CODE REVIEW

**Comment by CCorrado** [comment]
Looks like the tests failed:

```
======================================================================
ERROR: test_marriage_before_divorce (testMarriageBeforeDivorce.MarriageBeforeDivorce)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/jj1976/SSW555AgileMethodsCJBS/tests/testMarriageBeforeDivorce.py", line 16, in test_marriage_before_divorce
    for family in gedcom_data[0]:
TypeError: 'NoneType' object is not subscriptable
======================================================================
ERROR: test_marriage_descendants (testMarriageNotWithDescendants.MarriageNotWithDescendantsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/jj1976/SSW555AgileMethodsCJBS/tests/testMarriageNotWithDescendants.py", line 16, in test_marriage_descendants
    for individual in gedcom_data[1]:
TypeError: 'NoneType' object is not subscriptable
----------------------------------------------------------------------
Ran 34 tests in 0.090s
FAILED (errors=2)
```

File: tests/testMarriageNotWithDescendants.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+class MarriageNotWithDescendantsTests(unittest.TestCase):
+    """
+    Test Cases for U.S. 17
+    """          
+
+    def test_marriage_descendants(self):
+        """
+        Validate if a family member has married their own descendants
+        """
+        gedcom_data = parse_gedcom_file("../sampledata/us17testdata.ged")
```
Comment (CCorrado) [inline tests/testMarriageNotWithDescendants.py]: should just be `./` instead of `../`

File: tests/testMarriageBeforeDivorce.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+class MarriageBeforeDivorce(unittest.TestCase):
+    """
+    Test Cases for U.S. 04
+    """
+    
+    def test_marriage_before_divorce(self):
+        """
+        Validate if divorce date has happened before marriage date
+        """
+        gedcom_data = parse_gedcom_file("../sampledata/us04testdata.ged")
```
Comment (CCorrado) [inline tests/testMarriageBeforeDivorce.py]: should just be `./` instead of `../`

**Comment by rsukitha** [comment]
Thanks for letting me know Chris . I will change it now and check in

**Comment by CCorrado** [comment]
```
FAIL: test_marriage_descendants (testMarriageNotWithDescendants.MarriageNotWithDescendantsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/jj1976/SSW555AgileMethodsCJBS/tests/testMarriageNotWithDescendants.py", line 19, in test_marriage_descendants
    self.assertTrue(gedcom_data[1].get(individual).validate_spouse(id))
AssertionError: False is not true
```

**Pull request merged**